grails-app/init/org/pih/warehouse/BootStrap.groovy

```diff
@@ -605,8 +607,18 @@ class BootStrap {
             liquibase = new Liquibase('views/drop-all-views.xml', new ClassLoaderResourceAccessor(), database)
             liquibase.update(null as Contexts, new LabelExpression());
 
-            // FIXME Not good that we'll need to update this on subsequent versions so this needs some more thought
-            List previousChangelogVersions = ["0.5.x", "0.6.x", "0.7.x", "0.8.x"]
+            // Get all directories from /migrations
+            PathMatchingResourcePatternResolver resolver = new PathMatchingResourcePatternResolver()
+            Resource[] resources = resolver.getResources("file:grails-app/migrations/*")
+
+            // Find directories with names matching current versions pattern
+            List<String> changelogVersions = resources
+                    .collect { it.filename }
+                    .findAll { it.matches("[0-9]{1,3}.[0-9]{1,3}.x") }
+                    .reverse()
+
+            // Exclude the newest changelog version, this one should be run separately
+            List<String> previousChangelogVersions = changelogVersions.size() ? changelogVersions.tail() : []
```

Comment:
Also, are we sure that the resource resolver will return the directories in the right order every time. Or that findAll will? Or should we add an explicit alphanumeric sort to make sure we get the list back in the proper order.

Reply:
Directories are listed in the same order as they are in the /migrations directory. I created 0.4.x, 1.0.x, and 1.1.x to see how they will behave - each of those directories was placed in the right order when I created them:
![image](https://github.com/openboxes/openboxes/assets/83239466/73a6f6ad-5204-4ba3-b558-a0b4876e3e44)
0.4.x is the first of the list, and the 1.0.x and 1.1.x are at the end of the list, so because of that I am getting the list in the right order (from 0.4.x to 1.1.x).
So this is a list of changelog versions:
![image](https://github.com/openboxes/openboxes/assets/83239466/29517fef-c6fb-43da-8956-6f939c0ed87c)
And the tail of it:
![image](https://github.com/openboxes/openboxes/assets/83239466/35538e96-87a6-46be-ac4d-58319a488b5a)